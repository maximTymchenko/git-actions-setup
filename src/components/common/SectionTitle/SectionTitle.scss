@import '../../../styles/vars';

.title {
  font-family: 'OktaNeue-Black';
  color: $text;
  // margin-bottom: 25px;
  font-weight: 900;
  @include main__padding;
  padding-left: 0;

  @media screen and (max-width: $mobile) {
    padding: 0;
  }
}
.title--primary {
  font-size: 32px;

  @media screen and (max-width: $mobile) {
    padding: 0;
    font-size: 26px;
  }
}
.title--secondary {
  margin-top: 130px;
  max-width: 500px;
  font-size: 28px;

  @media screen and (max-width: $tablet) {
    margin-top: 100px;
  }

  @media screen and (max-width: $mobile) {
    margin-top: 40px;
    font-size: 19px;
  }
}

.title--main {
  margin-top: 150px;
  background: linear-gradient(
    to left,
    $brand__primary 30%,
    $secondary__dark 100%
  );
  font-size: 52px;
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;

  @media screen and (max-width: 850px) {
    margin-top: 180px;
  }

  @media screen and (max-width: $tablet) {
    margin-top: 50px;
    font-size: 44px;
  }

  @media screen and (max-width: $mobile) {
    font-size: 30px;
  }
}

.title--lighter {
  max-width: 700px;
  margin-top: 40px;
  background: linear-gradient(
    to left,
    $brand__primary 35%,
    $secondary__dark 80%
  );
  font-size: 56px;
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;

  @media screen and (max-width: $tablet) {
    margin-top: 0px;
  }

  @media screen and (max-width: $mobile) {
    font-size: 30px;
  }
}

.title--indent {
  max-width: 700px;
  margin-top: 0px;
  background: -o-linear-gradient(
    right,
    $brand__primary 35%,
    $secondary__dark 80%
  );
  background: -webkit-gradient(
    linear,
    right top,
    left top,
    color-stop(35%, $brand__primary),
    color-stop(80%, $secondary__dark)
  );
  background: linear-gradient(
    to left,
    $brand__primary 35%,
    $secondary__dark 80%
  );
  font-size: 56px;
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;

  @media screen and (max-width: $tablet) {
    margin-top: 0px;
  }

  @media screen and (max-width: $mobile) {
    font-size: 44px;
  }
}

.title--relevant {
  max-width: 530px;
  font-family: 'OktaNeue-SemiBold';
  font-size: 24px;
  font-weight: 600;
  line-height: 1.5;

  @media screen and (max-width: $mobile) {
    font-size: 22px;
  }
}

.title--city {
  font-size: 24px;
  letter-spacing: 0.4px;

  @media screen and (max-width: $mobile) {
    margin-top: 10px;
  }
}

.title--section {
  margin-top: 10px;
  font-family: 'OktaNeue-Heavy';
  font-weight: 900;
  line-height: 90px;
  margin-right: 25px;
  margin-bottom: 10px;
  margin-top: 0px;
  font-size: 68px;
  background: linear-gradient(
    to left,
    $brand__primary 50%,
    $secondary__dark 95%
  );
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;
  display: block;

  @media screen and (max-width: $desktop) {
    font-size: 64px;
  }

  @media screen and (max-width: $tablet) {
    margin-top: 0px;
    font-size: 60px;
  }

  @media screen and (max-width: $small__tablet) {
    font-size: 50px;
  }

  @media screen and (max-width: $large__mobile) {
    line-height: 50px;
    font-size: 39px;
  }
}

.title--small {
  margin-bottom: 0px;
  background: -o-linear-gradient(
    right,
    $brand__primary 50%,
    $secondary__dark 95%
  );
  background: -webkit-gradient(
    linear,
    right top,
    left top,
    color-stop(50%, $brand__primary),
    color-stop(95%, $secondary__dark)
  );
  background: linear-gradient(
    to left,
    $brand__primary 50%,
    $secondary__dark 95%
  );
  font-family: 'OktaNeue-Light';
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;
  font-size: 26px;
  font-weight: 300;
  display: block;

  @media screen and (max-width: $mobile) {
    margin-top: 0px;
    line-height: 25px;
    font-size: 26px;
  }
}

.title--bold {
  margin-bottom: 0px;
  background: -o-linear-gradient(
    right,
    $brand__primary 50%,
    $secondary__dark 95%
  );
  background: -webkit-gradient(
    linear,
    right top,
    left top,
    color-stop(50%, $brand__primary),
    color-stop(95%, $secondary__dark)
  );
  background: linear-gradient(
    to left,
    $brand__primary 50%,
    $secondary__dark 95%
  );
  font-family: 'OktaNeue-Light';
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;
  font-size: 26px;
  font-weight: bold;
  display: block;

  @media screen and (max-width: $mobile) {
    margin-left: 7px;
    margin-right: 7px;
  }
}

.title--brackets {
  position: relative;
  display: block;

  &::before {
    font-family: 'OktaNeue-Thin';
    content: '{';
    font-weight: 100;
    font-size: 88px;

    @media screen and (max-width: $desktop) {
      font-size: 64px;
    }

    @media screen and (max-width: $tablet) {
      font-size: 60px;
    }

    @media screen and (max-width: $small__tablet) {
      font-size: 50px;
    }

    @media screen and (max-width: $large__mobile) {
      font-size: 40px;
    }
  }

  &::after {
    font-family: 'OktaNeue-Thin';
    content: '}';
    font-size: 88px;
    font-weight: 100;

    @media screen and (max-width: $desktop) {
      font-size: 64px;
    }

    @media screen and (max-width: $tablet) {
      font-size: 60px;
    }

    @media screen and (max-width: $small__tablet) {
      font-size: 50px;
    }

    @media screen and (max-width: $large__mobile) {
      font-size: 40px;
    }

    // @media screen and (max-width: $mobile) {
    //     font-size: 32px;
    // }
  }
}
