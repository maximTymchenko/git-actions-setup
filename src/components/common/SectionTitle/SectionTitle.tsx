import React, { FC } from 'react';
import './SectionTitle.scss';

interface ITitleProps {
    className?: string;
    title: string;
    type: any;
}

const SectionTitle: FC = (props: ITitleProps) => {
    return (
        <h2 data-testid="section-title" className={`title title--${props.type}`}>
            {props.title}
        </h2>
    );
};

export default SectionTitle;
