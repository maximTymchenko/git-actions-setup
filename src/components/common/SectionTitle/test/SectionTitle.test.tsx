import React from 'react';
import SectionTitle from '../SectionTitle';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<SectionTitle title="test"></SectionTitle>);
    expect(getByTestId('section-title')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<SectionTitle title="test"></SectionTitle>).toJSON();
    expect(tree).toMatchSnapshot();
});
