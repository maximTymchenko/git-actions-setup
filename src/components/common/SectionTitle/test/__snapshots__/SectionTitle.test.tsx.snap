// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<h2
  className="title title--undefined"
  data-testid="section-title"
>
  test
</h2>
`;
